src/crimson/osd/pg.cc

```diff
@@ -505,7 +506,7 @@ void PG::on_active_actmap()
       }).then([this, trimmed=to_trim] {
         logger().debug("{}: trimmed snap={}", *this, trimmed);
       });
-    }).finally([this] {
+    }).finally([this, pg_life_extender=shared_from_this()] {
```

Comment:
Nit:

Seems to me the other places don't need the life extender as long as the finally() has.

And it will be cleaner (at least for me) if the indention can be like, e.g.:
```c++
std::ignore = seastar::do_until(
  [this] {
    ...
  },
  [this] {
    ...
  }
// I think this will be easier to see that finally() is chained after do_until()
).finally([this, pg_life_extender=shared_from_this()] { 
  ...
});
```